pkg/provider/remote/remote.go

```diff
@@ -0,0 +1,352 @@
+// Copyright Nitric Pty Ltd.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package remote
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+
+	"github.com/pkg/errors"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+
+	"github.com/nitrictech/cli/pkg/output"
+	"github.com/nitrictech/cli/pkg/project"
+	"github.com/nitrictech/cli/pkg/provider/types"
+	deploy "github.com/nitrictech/nitric/core/pkg/api/nitric/deploy/v1"
+	v1 "github.com/nitrictech/nitric/core/pkg/api/nitric/v1"
+)
+
+type remoteDeployment struct {
+	proj      *project.Project
+	stackName string
+	provider  string
+	conn      *grpc.ClientConn
+}
+
+type stackSummary struct {
+	Name             string `json:"name"`
+	Deployed         bool   `json:"deployed"`
+	LastUpdate       string `json:"lastUpdate,omitempty"`
+	UpdateInProgress bool   `json:"updateInProgress"`
+	ResourceCount    *int   `json:"resourceCount,omitempty"`
+	URL              string `json:"url,omitempty"`
+}
+
+var _ types.Provider = &remoteDeployment{}
+
+func New(p *project.Project, name, provider string, envMap map[string]string, opts *types.ProviderOpts) (types.Provider, error) {
+	conn, err := grpc.Dial("127.0.0.1:50051", grpc.WithTransportCredentials(insecure.NewCredentials()), grpc.WithBlock())
+	if err != nil {
+		return nil, err
+	}
+
+	return &remoteDeployment{
+		proj:      p,
+		stackName: name,
+		provider:  provider,
+		conn:      conn,
+	}, nil
+}
+
+func (p *remoteDeployment) AskAndSave() error {
+	return errors.New("not supported on remote deployment servers")
+}
+
+func (p *remoteDeployment) SupportedRegions() []string {
+	return []string{}
+}
+
+func execNameFromFunctions(funcs map[string]project.Function, target *project.ScheduleTarget) string {
+	switch target.Type {
+	case "function":
+		return target.Name
+
+	case "queue":
+		return ""
+
+	default:
+		for _, fv := range funcs {
+			for _, topicName := range fv.ComputeUnit.Triggers.Topics {
+				if topicName == target.Name {
+					return fv.Name
+				}
+			}
+		}
+
+		return ""
+	}
+}
```

Comment:
This might seem like a bit of over duplication but I think this logic belongs directly with CodeConfig as another `ToDeployUpRequest` or `ToDeployUpResources`. To save translating legacy concepts like this.

For more complex translation like building API documentation we can use shared methods to avoid duplication in these cases.